plugins/upstash/upstash.go

```diff
@@ -0,0 +1,26 @@
+package upstash
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+)
+
+func UpstashCLI() schema.Executable {
+	return schema.Executable{
+		Name:      "Upstash CLI", 
+		Runs:      []string{"upstash"},
+		DocsURL:   sdk.URL("https://github.com/upstash/cli"), 
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotForHelpOrVersion(),
+			needsauth.NotWithoutArgs(),
+			needsauth.NotWhenContainsArgs("auth","login"),
```

Comment:
```suggestion
			needsauth.NotWhenContainsArgs("auth"),
```
I think `auth` is enough because it seems to cover both `login` and `logout`: https://github.com/upstash/cli#usage
The previous formulation means to only skip provisioning if commands such like `upstash auth login`

Reply:
Looks addressed now.